pkg/registries/docker/docker.go

```diff
@@ -22,8 +22,22 @@ import (
 )
 
 const (
-	registryTimeout  = 5 * time.Second
-	repoListInterval = 10 * time.Minute
+	// registryDialerTimeout is the net.Dialer timeout of the client transport.
+	// It limits the time the dialer attempts the connection. The timeout is
+	// chosen as a small value to prevent unavailable registries from blocking
+	// image scanning.
+	registryDialerTimeout = 5 * time.Second
+	// registryResponseTimeout is the response header timeout of the client transport.
+	// It limits the time to wait for a server's response headers after fully
+	// writing the request.
+	registryResponseTimeout = 60 * time.Second
+	// registryClientTimeout is used as http.Client.Timeout for the registry's HTTP
+	// client and hence includes everything from connection to reading the
+	// response body. The timeout has been chosen rather arbitrarily, it is
+	// probably less harm in waiting a bit longer than in aborting early a
+	// request that is about to succeed.
```

Comment:
I'd like to see this assumption be validated in some way before we enable these new settings for _all_ registry connections. I do think that "waiting a bit longer" has more impact than we ideally want to (specifically when our image integration matching is lackluster, specifically when dealing with autogenerated image integrations).

I remember we initially wanted to enabled this for CI only and observe internally what kind of an effect this has based on some metrics (which we would need to add here as well). What happened with that?

Reply:
> I'd like to see this assumption be validated in some way before we enable these new settings for all registry connections. I do think that "waiting a bit longer" has more impact than we ideally want to (specifically when our image integration matching is lackluster, especially when dealing with autogenerated image integrations).

I think it will be hard to prove the negative ("there is definitively no negative impact in any scenario"). But I find it hard to imagine how this could be a catastrophic change. I think we should avoid speculating on hypothetical worst case scenarios. Do you have a concrete scenario in mind that you are worried about?

> I remember we initially wanted to enabled this for CI only and observe internally what kind of an effect this has based on some metrics (which we would need to add here as well). What happened with that?

As proposed in https://docs.google.com/document/d/1gvaHq7_iYAXAJSi9nfR6Ppq4lEnPtS9kAGdZ25VP9bQ/edit#heading=h.x1u2pndrgww9 my intention was always to generally increase the timeouts, although we can use different values in CI. The follow up PR makes this possible by exposing the time out values as env vars.

Reply:
>Do you have a concrete scenario in mind that you are worried about?

The one I have in mind is when auto-generated image integrations are being used, which are duplicated for each cluster, and running into the scenario [of signature fetching here](https://github.com/stackrox/stackrox/blob/master/pkg/images/enricher/enricher_impl.go#L748-L783). In case the requests now take significantly longer, this will have an impact on the duration of the enrichment. Same applies for [metadata here](https://github.com/stackrox/stackrox/blob/master/pkg/images/enricher/enricher_impl.go#L368-L389) albeit metadata is cached and it will only take longer during the first scan.

Another unrelated thing: this now only touches AFAICT the timeout settings for Central accessing registries (i.e. to fetch metadata or signatures); is there any plan to look at the timeout values scanner uses to reach out to the registries as well? I would assume they use Central's registry implementation, but not entirely sure if they actually do.

>The follow up PR makes this possible by exposing the time out values as env vars.

Ah, gotcha, I didn't see the stack of PRs.

Reply:
> In case the requests now take significantly longer
Yes, but in what scenario does every image integration actually run into the max timeout? I think if we can prevent long timeouts for unavailable registries (see dial timeout), then it should be quite unlikely to have a significant impact. Although the proxy issue that you mentioned could be problematic for this reason.

> In case the requests now take significantly longer, this will have an impact on the duration of the enrichment.

To play devil's advocate: Why is this is a (big) problem?

> Another unrelated thing: this now only touches AFAICT the timeout settings for Central accessing registries (i.e. to fetch metadata or signatures); is there any plan to look at the timeout values scanner uses to reach out to the registries as well? I would assume they use Central's registry implementation, but not entirely sure if they actually do.

Plans yes, but haven't done that yet.

Reply:
I ended up reducing the timeout values down to 10 seconds because of the issue with proxies not respecting the connection timeout of the dialer.